### PR TITLE
Use `ReferenceTestWorldState` instead of `ToyWorld`

### LINKED ISSUE #969: Replace `ToyWorld` with `SimpleWorld` for `ReplayTests`
The goal here is to use the `SimpleWorld` implementation which is part of BESU.  Since we have had some problems with `ToyWorld` this should hopefully resolve them.

I have prototyped this already, and the `ReplayTests` at least it is possible to do.  Potentially, for other tests it might also be possible.

### PR DESCRIPTION
**Pull request by DavePearce**
<strike>This puts in place a temporary fix which converts `ToyWorld` over into `ReferenceTestWorldState` to use when executing transactions.  The conversion could be done more elegantly, but the goal here is to get something out for people to use ASAP.  Eventually, we would then remove `ToyWorld` altogether.</strike>

**UPDATE**: to get the fix out there more quickly, it was cherry-picked into #1027.  What's left of this PR is relatively straightforward.  It just updates some of the replay files so they now all include tx result / blockhash information.  It also marks all tests which pass tx result checking.  Only a small number remain which fail for various reasons (e.g. problems with the balance or `nonce`).  Those issues will be resolved separately.

### CODE REVIEW

**Comment by DavePearce** [comment]
Will wait for the PR for  #1006 to be merged before merging this, since that now contains a commit from this PR (which was cherry-picked).

**Review by letypequividelespoubelles — APPROVED** [review]

**Pull request merged**
